src/transformers/models/whisper/generation_whisper.py

```diff
@@ -231,7 +232,7 @@ def _extract_token_timestamps(self, generate_outputs, alignment_heads, time_prec
 
         # Perform dynamic time warping on each element of the batch.
         for batch_idx in range(batch_size):
-            if num_frames is not None and isinstance(num_frames, (tuple, list, np.ndarray)):
+            if num_frames is not None and isinstance(num_frames, (tuple, list, np.ndarray, torch.Tensor)):
                 matrix = weights[batch_idx, ..., : num_frames[batch_idx] // 2]
```

Comment:
Since here we slice `weights` (torch tensor) by `num_frames` (which should be most efficient if both are tensors no?)